### DISCUSSION TITLE
How to support inband mode instead of RFC2833?

### DISCUSSION THREAD

**Discussion by mgsray** [sipsorcery-org/sipsorcery#553] [Q&A]
I have a requirement to initial a call with inband mode, in order to use DTMF during the call. Because the answer side only supports inband mode to read DTMF.

**Comment by sipsorcery** [comment]
Inband means the DTMF tones will be sent in the audio stream and are from en external point of view the DTMF packets are indistinguisahble from any other audio packet. To extract the DTMF tones you need to process the received audio stream and extract them.

The sipsorcery library doesn't have any logic to extract DTMF tones from an audio stream. NAudio might. If not there are lots of algorithms around.

**Reply by mgsray** [reply]
Thanks for reply. 
I use sipsorcery as a sip client to a freeswitch sip server, and client doesn't have to extract DTMF tones. The scenario is, I use the client to get audio stream which might has DTMF, I want to transfer the audio stream and extract them by server. 
Is that possible?

**Comment by sipsorcery** [comment]
Sounds like you don't need to do anything. If the audio stream your client is sending already has DTMF tones within it then there's nothing else for the client to do. You do need to make sure your FreeSWITCH server has the right DTMF option configured.
